flink-runtime/src/main/java/org/apache/flink/runtime/metrics/dump/MetricQueryService.java

```diff
@@ -73,60 +71,50 @@ public String filterCharacters(String input) {
 
 	private final long messageSizeLimit;
 
-	public MetricQueryService(long messageSizeLimit) {
+	public MetricQueryService(RpcService rpcService, String endpointId, long messageSizeLimit) {
+		super(rpcService, endpointId);
 		this.messageSizeLimit = messageSizeLimit;
 	}
 
 	@Override
-	public void postStop() {
+	public CompletableFuture<Void> onStop() {
 		serializer.close();
+		return CompletableFuture.completedFuture(null);
 	}
 
-	@Override
-	public void onReceive(Object message) {
-		try {
```

Comment:
how does the stack trace look like if any of the methods throw an exception?

Reply:
All of `addMetric` `removeMetric` and `queryMetrics` don't throw Exception. If it is a fatal, the rpcendpoint will down. Any exception thrown from the methods should be the same as a exception throw from `AkkaRpcActor#handleRpcInvocation`.